runtime/index.d.ts

```diff
@@ -35,3 +36,22 @@ declare module "@sapper/service-worker" {
 	export const shell: string[];
 	export const routes: Array<{ pattern: RegExp }>;
 }
+
+declare module "@sapper/common" {
+	export interface PreloadContext {
+		fetch: (url: string, options?: any) => Promise<any>;
+		error: (statusCode: number, message: Error | string) => void;
+		redirect: (statusCode: number, location: string) => void;
+	}
+
+	export interface PreloadPage {
+		host: string;
+		path: string;
+		params: Record<string, string>;
+		query: Record<string, string | string[]>;
+	}
+
+	export interface PreloadFunction {
```

Comment:
How/where would this be used?

Reply:
```ts
export const preload: PreloadFunction = (page, session) => {
    this... // <- works, too
};
```

Reply:
I think we should require that it return an object (rather than `any`). I assume returning say a string is not supported.

Reply:
Yeah, we set the return type to `object | Promise<object>` in https://github.com/sveltejs/sapper/pull/1528